vendor/github.com/dgrijalva/jwt-go/map_claims.go

```diff
@@ -25,49 +27,37 @@ func (m MapClaims) VerifyAudience(cmp string, req bool) bool {
 			}
 			aud = append(aud, vs)
 		}
-	case string:
-		aud = append(aud, v)
-	default:
-		return false
 	}
 	return verifyAud(aud, cmp, req)
 }
 
 // Compares the exp claim against cmp.
 // If required is false, this method will return true if the value matches or is unset
 func (m MapClaims) VerifyExpiresAt(cmp int64, req bool) bool {
-	exp, ok := m["exp"]
-	if !ok {
-		return !req
-	}
-	switch expType := exp.(type) {
+	switch exp := m["exp"].(type) {
```

Comment:
Looks like the official repository does not have the security fix that the `orm3tech-oss/jwt-go` fork has; switching to `github.com/golang-jwt/jwt` removes these changes, which seem to be related to that security fix: https://github.com/form3tech-oss/jwt-go/compare/v3.2.3...v3.2.4

Reply:
I believe golang-jwt/jwt v3.2.1 fixed CVE-2020-26160 with different code: https://github.com/golang-jwt/jwt/commit/0f726ea0e7256f9056da372f822bd534f9d1dc5f

Reply:
The equivalent to that one is in https://github.com/form3tech-oss/jwt-go/commit/bb5e6d8f35bea762d2564725ef646dd6e6a57799 for the form3tech repository. Looks like v3.2.4 is fixing a different security issue?

Reply:
Correct. I suspect no CVE has been requested for the security fix in  form3tech-oss/jwt-go v3.2.4. If it's verified to be a security issue, a CVE should probably be requested (both for form3tech-oss/jwt-go, and for  github.com/golang-jwt/jwt / dgrijalva/jwt-go)

Reply:
Shall we merge those two PRs while we wait for the situation with this new vulerability to be resolved upstream or do you think the workaround in this PR is a better fix for right now?

Reply:
My concern with https://github.com/distribution/distribution/pull/3465 is that it's pulling in 5.6k lines of code that's unrelated to the security fix (and which would have to be reviewed), with the goal to pull in a security fix that's < 50 lines. And (worse?) switching to a non-official fork (which, based on the `go.mod` addition, is at least somewhat "dubious" on quality).

From that perspective, I'd rather take the official fork (_ideally_ with https://github.com/golang-jwt/jwt/pull/40, if that's confirmed to be an issue), as is done in this PR.

For the `main` branch, I don't think there's an direct urgency to get it merged, unless a `v3.0.0` release is pending.